### PR TITLE
OSDOCS-3602: Document image registry distribution across availability zones

### PR DESCRIPTION
**Pull request by bmcelvee**
Version(s):
4.11

Issue:
https://issues.redhat.com/browse/OSDOCS-3602

Link to docs preview:
http://file.rdu.redhat.com/~bmcelvee/20221107/registry/configuring-registry-operator.html#registry-operator-distribution-across-availability-zones_configuring-registry-operator

Additional information:

### CODE REVIEW

File: registry/configuring-registry-operator.adoc [hunk 1]
```diff
@@ -51,6 +40,13 @@ However, the `managementState` of the Image Registry Operator alters the behavio
 
 include::modules/registry-removed.adoc[leveloffset=+2]
 
+include::modules/registry-operator-distribution-across-availability-zones.adoc[leveloffset=+1]
```
Comment (bmcelvee) [inline registry/configuring-registry-operator.adoc:43]: New content

**Comment by bmcelvee** [comment]
Hi @xiuwang, would you mind giving this PR QE review please? Thank you!

File: modules/registry-operator-distribution-across-availability-zones.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+The Image Registry Operator defaults to:
+[source,yaml]
+----
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        docker-registry: default
```
Comment (xiuwang) [inline modules/registry-operator-distribution-across-availability-zones.adoc:20]: Not in alphabetical order
```
  topologySpreadConstraints:
  - labelSelector:
      matchLabels:
        docker-registry: default
    maxSkew: 1
    topologyKey: kubernetes.io/hostname
    whenUnsatisfiable: DoNotSchedule
  - labelSelector:
      matchLabels:
        docker-registry: default
    maxSkew: 1
    topologyKey: node-role.kubernetes.io/worker
    whenUnsatisfiable: DoNotSchedule
  - labelSelector:
      matchLabels:
        docker-registry: default
    maxSkew: 1
    topologyKey: topology.kubernetes.io/zone
    whenUnsatisfiable: DoNotSchedule
```

File: modules/registry-operator-distribution-across-availability-zones.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - maxSkew: 1
+    labelSelector:
+      matchLabels:
+        docker-registry: default
+    topologyKey: node-role.kubernetes.io/worker
+    whenUnsatisfiable: DoNotSchedule
+----
+
+A cluster administrator can override the default `topologySpreadConstraints` by configuring `ImageRegistry` Spec. In that case, only the constraints you provided apply.
```
Comment (xiuwang) [inline modules/registry-operator-distribution-across-availability-zones.adoc:35]: Better to change s/`ImageRegistry`/`configs.imageregistry.operator.openshift.io/cluster`

Comment (xiuwang) [reply modules/registry-operator-distribution-across-availability-zones.adoc:35]: Could we add this bug scenario, https://bugzilla.redhat.com/show_bug.cgi?id=2067060#c3  ?
When there is no zone label setting, the topologySpreadConstraints will be default to
```
  - maxSkew: 1 
    topologyKey: kubernetes.io/hostname 
    whenUnsatisfiable: DoNotSchedule
    labelSelector:
      matchLabels:
        docker-registry: default
  - maxSkew: 1 
    labelSelector:
      matchLabels:
        docker-registry: default
    topologyKey: node-role.kubernetes.io/worker
    whenUnsatisfiable: DoNotSchedule
```

File: modules/registry-operator-distribution-across-availability-zones.adoc (hunk 2 above)
Comment (bmcelvee) [reply modules/registry-operator-distribution-across-availability-zones.adoc:29]: Thank you! Updated.

File: modules/registry-operator-distribution-across-availability-zones.adoc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        docker-registry: default
+  - maxSkew: 1
+    labelSelector:
+      matchLabels:
+        docker-registry: default
+    topologyKey: node-role.kubernetes.io/worker
+    whenUnsatisfiable: DoNotSchedule
```
Comment (xiuwang) [inline modules/registry-operator-distribution-across-availability-zones.adoc:53]: @bmcelvee Sorry for my mistake, it should be 
```
 topologySpreadConstraints:
  - labelSelector:
      matchLabels:
        docker-registry: default
    maxSkew: 1
    topologyKey: kubernetes.io/hostname
    whenUnsatisfiable: DoNotSchedule
  - labelSelector:
      matchLabels:
        docker-registry: default
    maxSkew: 1
    topologyKey: node-role.kubernetes.io/worker
    whenUnsatisfiable: DoNotSchedule
```

File: modules/registry-operator-distribution-across-availability-zones.adoc [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  - labelSelector:
+      matchLabels:
+        docker-registry: default
+    maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+----
+
+The Image Registry Operator defaults to the following when deployed without a zone related topology constraint, which applies to bare metal and vSphere instances:
```
Comment (lpettyjo) [inline modules/registry-operator-distribution-across-availability-zones.adoc:37]: Should "zone related" be hyphenated?

**Review by lpettyjo — COMMENTED** [review]
Otherwise, LGTM.

File: modules/registry-operator-distribution-across-availability-zones.adoc (hunk 4 above)
Comment (xiuwang) [reply modules/registry-operator-distribution-across-availability-zones.adoc:53]: @bmcelvee Above comment need address, thanks

Comment (bmcelvee) [reply modules/registry-operator-distribution-across-availability-zones.adoc:53]: Yes, thanks! I'm updating it now.

Comment (bmcelvee) [reply modules/registry-operator-distribution-across-availability-zones.adoc:54]: Updated!

**Comment by xiuwang** [comment]
LGTM
Thanks @bmcelvee

**Pull request merged**